### PR TITLE
Add Class class

### PR DESCRIPTION
**Pull request by adiaaida**

### CODE REVIEW

File: Core.Tests/CharacterTests.cs [hunk 1]
```diff
@@ -11,5 +11,23 @@ public void NameTest()
             var c = new Character {Name = "Sir Testy McTestington"};
             c.Name.Should().Be("Sir Testy McTestington");
         }
+
+        [Test]
+        public void ClassTest()
+        {
+            var sorcerer = new Class {
+                Name = "Sorcerer",
+                HitDie = 6,
```
Comment (jonfortescue) [inline Core.Tests/CharacterTests.cs:20]: Missing `SkillRanks` property.

File: Core/Class.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        // Maybe should be an enum?
+        public string Name { get; set; }
+
+        public int HitDie { get; set; }
+
+        public int SkillRanks { get; set; }
+
+        public float BaseAttackBonus { get; set; }
+
+        public float FortitudeSaveBase { get; set; }
```
Comment (jonfortescue) [inline Core/Class.cs:16]: I'm not sure how we're planning to use Lua, but I think we could simplify these with some lambdas e.g.

```csharp
public Func<int, int> FortitudeSave { get; set; }
```

And then we could have a struct to represent the standard fast/slow save progressions:
```csharp
public struct SaveProgressions
{
	public static Func<int, int> FastProgression => (level => level / 2 + 2);
	public static Func<int, int> SlowProgression => (level => level / 3);
}
```

That would make your sorcerer declaration up top:
```csharp
Class sorceror = new Class
	{
		Name = "Sorceror",
		HitDie = 6,
		SkillRanks = 2,
		BaseAttackBonus = 1f/3f,
		FortitudeSave = SaveProgressions.SlowProgression,
		ReflexSave = SaveProgressions.SlowProgression,
		WillSave = SaveProgressions.FastProgression,
	};
```

File: Core/Class.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    public class Class
+    {
+        // Maybe should be an enum?
+        public string Name { get; set; }
+
+        public int HitDie { get; set; }
+
+        public int SkillRanks { get; set; }
+
+        public float BaseAttackBonus { get; set; }
```
Comment (jonfortescue) [inline Core/Class.cs:14]: Similar to the save progression comment below, I think BAB should probably be a lambda.

**Comment by adiaaida** [comment]
Eh. Abandoning after last night's discussions.

**Pull request closed without merging**